app/routes/groups_.$groupId_.events_.new.tsx

```diff
@@ -7,6 +7,11 @@ import { Card } from '~/components/ui/containers';
 import { Input, TextArea } from '~/components/ui/forms';
 import { Button } from '~/components/ui/button';
 import { H1 } from '~/components/ui/headers';
+import { requireUserSession } from '~/modules/session/session.server';
+
+export async function loader({ request }: LoaderFunctionArgs) {
```

Comment:
As a next step, we could also check here that the user has "admin" access to the group before we allow them to create a new event for that group!